tests/cases/expr/let.slint

```diff
@@ -0,0 +1,72 @@
+// Copyright © SixtyFPS GmbH <info@slint.dev>
+// SPDX-License-Identifier: GPL-3.0-only OR LicenseRef-Slint-Royalty-free-2.0 OR LicenseRef-Slint-Software-3.0
+
+export component TestCase {
+    public function test_simple() -> int {
+        let a = 1;
+
+        return a;
+    }
+
+    public function test_shadowing() -> int {
+        let a = 1;
+        let a = a + 1;
+
+        return a;
+    }
+
+    public function test_shadowing_scope() -> int {
+        let a = 1;
+
+        if (true) {
+            let a = a + 1;
+
+            return a;
```

Comment:
can we also have a version of the test without the return here?  (to make sure that the original value is not changed.